File: algorithms/boyer-moore-majority-vote/python/boyer_moore_majority_vote/boyer_moore_majority_vote.py
def get_majority_element(elements):
    candidate = None
    count = 0

    for element in elements:
        if count == 0:
            candidate = element

        if candidate == element:
            count += 1
        else:
            count -= 1

    return candidate


elements = [1, 2, 2, 2, 2, 2, 2, 2, 2, 3, 4, 5, 6, 7, 8, 9, 2]
assert(get_majority_element(elements) == 2)

File: algorithms/boyer_moore_majority_vote/ruby/boyer_moore_majority_vote/boyer_moore_majority_vote.rb
def get_majority_element(elements)
    candidate = nil 
    count = 0

    elements.each do |element|
        candidate = count == 0 ? candidate = element : candidate
        count += candidate == element ? +1 : -1
    end

    candidate
end

elements = [1, 1, 1, 1, 3, 5, 7, 6]
puts(get_majority_element(elements) == 1)